src/realm/sync/noinst/sync_schema_migration.cpp

```diff
@@ -117,9 +117,18 @@ void perform_schema_migration(DB& db)
     repl.set_write_validator_factory(nullptr);
 
     // Delete all tables (and their columns).
-    const bool ignore_backlinks = true;
     for (const auto& tk : tr->get_table_keys()) {
-        tr->remove_table(tk, ignore_backlinks);
+        auto cur_table = tr->get_table(tk);
+        // First delete the columns linking to this table.
+        cur_table->for_each_backlink_column([&](ColKey backlink_col_key) {
```

Comment:
To me it seems simpler to remove all columns from all tables (4d3c1d432a2a96534d2bec5893b22c43d8c9253e) before removing any tables rather than removing just the incoming links.

Reply:
that will do it. you can push it to the branch